src/state/slices/tradeQuoteSlice/selectors.ts

```diff
@@ -723,8 +730,20 @@ export const selectIsAnyTradeQuoteLoading = createSelector(
 )
 
 export const selectIsAnyTradeQuoteLoaded = createSelector(
-  selectIsSwapperQuoteAvailable,
   selectHasUserEnteredAmount,
-  (isSwapperQuoteAvailable, hasUserEnteredAmount) =>
-    !hasUserEnteredAmount || Object.values(isSwapperQuoteAvailable).some(identity),
+  selectIsAnyTradeQuoteLoading,
+  selectIsAnySwapperQuoteAvailable,
+  (hasUserEnteredAmount, isAnyTradeQuoteLoading, isAnySwapperQuoteAvailable) => {
+    if (!hasUserEnteredAmount) {
+      return true
+    }
+
+    // If we're still loading, return true if there is any quote available
+    if (isAnyTradeQuoteLoading) {
+      return isAnySwapperQuoteAvailable
+    }
+
+    // Otherwise, an empty result should default to true to allow the app to stop loading state.
+    return true
```

Comment:
[preferably blocking]

The logic here makes total sense, but now the name of the selector is a bit off.

Maybe something like `selectAreTradeQuotesLoaded` or something.

Reply:
Nice catch, went with `selectShouldShowTradeQuoteOrAwaitInput` which is hopefully a better representation of what's happening c1fa8be3b9c23b89ea65cdb168367f70332b418a